File: Arbitration/MPArbitration/Model/CalculatorVariables.cs
using System.Text.Json.Serialization;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;

namespace MPArbitration.Model
{
    [Index(nameof(ServiceLine))]
    public class CalculatorVariable
    {
        [JsonPropertyName("id")]
        public int Id { get; set; } = 0;

        [JsonPropertyName("arbitrationFee")]
        public double ArbitrationFee { get; set; } = 0;

        [JsonPropertyName("chargesCapDiscount")]
        public double ChargesCapDiscount { get; set; } = 0;  // percentage

        [JsonPropertyName("createdBy")]
        [StringLength(60)]
        public string CreatedBy { get; set; } = "";

        [JsonPropertyName("createdOn")]
        public DateTime? CreatedOn { get; set; } = null;

        [JsonPropertyName("nsaOfferDiscount")]
        public double NSAOfferDiscount { get; set; } = 0;

        [JsonPropertyName("nsaOfferBaseValueFieldname")]
        [StringLength(255)]
        public string NSAOfferBaseValueFieldname { get; set; } = "";

        [JsonPropertyName("offerCap")]
        public double OfferCap { get; set; } = 0;  // e.g. 35,000.00

        [JsonPropertyName("offerSpread")]
        public double OfferSpread { get; set; } = 0;   // percentage

        [JsonPropertyName("serviceLine")]
        [StringLength(20)]
        public string ServiceLine { get; set; } = "";  // e.g. IOM
    }
}

File: Arbitration/MPArbitration/ClientApp/src/app/model/calculator-variables.ts
export class CalculatorVariables {
    id = 0;
    arbitrationFee = 0;
    chargesCapDiscount = 0;  // percentage
    createdBy = '';
    createdOn:Date|undefined;
    nsaOfferDiscount = 0;
    nsaOfferBaseValueFieldname = ""; //fh80thPercentileExtendedCharges
    offerCap = 0;  // e.g. 35,000.00
    offerSpread = 0;   // percentage
    serviceLine = '';  // e.g. IOM

    constructor(obj?:any) {
        if(!obj)
            return;
        Object.assign(this, JSON.parse(JSON.stringify(obj)));
        if(obj.createdOn)
            this.createdOn = new Date(obj.createdOn);
        if(!this.nsaOfferBaseValueFieldname)
            this.nsaOfferBaseValueFieldname = 'fh80thPercentileExtendedCharges';
    }
}
